<a href="https://colab.research.google.com/github/JefferRua/MBA_EDA/blob/main/Taller_Final_EDA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Analisis Exploratorio de Datos
### Trabajo Final

### Jefferson Rua
### Xiomara Melina Ortiz
### Daniel Eduardo Castaño

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
pd.set_option('display.max_columns', None)

### Introducción del Dataset, pregunta a resolver y breve descripción de los objetivos que se buscan al realizar un EDA de la base

La base de datos que se va a analizar es el resultado de un sondeo realizado por la Cámara de Comercio de Cali en el marco de su programa "Cámara a la Calle". El objetivo de este programa era acercar la institución a los empresarios de poblaciones priorizadas, esta priorización se hizo teniendo en cuenta que la jurisdicción de la Cámara de Comercio incluye los municipios de:  Yumbo, Jamundí, Vijes, Dagua y La Cumbre y los empresarios de estos municipios no tienen una tasa de participación alta en los programas de la CCC, por tanto en el objetivo del programa era acercar la institución a los municipios de la jurisdicción y a algunos barrios de Cali identificados teniendo en cuenta su dinámica empresarial y baja participación en programas CCC.

En el marco de este programa se realizaron jornadas de ............ en las que se incluía una encuesta a los participantes de estas jornadas, el resultado de este sondeo:” Empresas Vibrantes" es el que se analizara a continuación. En primer lugar, se busca caracterizar a los empresarios que asistieron a las jornadas; en segundo lugar, se busca 'mapear' las necesidades empresariales de estos y diseñar programas de formación acordes con sus necesidades. 


In [ ]:
####Cargando base de datos con todos los condicionales
url="https://raw.githubusercontent.com/JefferRua/MBA_EDA/main/Resultados%20Sondeo%20Empresas%20Vibrantes%202022%20CCC.csv"
df=pd.read_csv(url,encoding='latin-1',sep=";", thousands='.')
df

Municipio Sexo del propietario del negocio o empresa:  \
0     Alameda                                       Mujer   
1     Alameda                                      Hombre   
2     Alameda                                      Hombre   
3     Alameda                                      Hombre   
4     Alameda                                      Hombre   
..        ...                                         ...   
733     Yumbo                                      Hombre   
734     Yumbo                                       Mujer   
735     Yumbo                                       Mujer   
736     Yumbo                                      Hombre   
737     Yumbo                                       Mujer   

     ¿Qué edad tiene el propietario del negocio o empresa? Registrado CCC  \
0                                                   37                 No   
1                                                   57                 Sí   
2                                                   40                 Sí   
3                                                   60                 Sí   
4                                                   72                 Sí   
..                                                 ...                ...   
733                                                 35                 Sí   
734                                                 43                 Sí   
735                                                 34                 No   
736                                                 34                 No   
737                                                 32                 Sí   

    FECHA_MATRICULA  AÑOS_ANTIGUEDAD AÑOS ANTIGU V2  \
0               NaN              NaN            NaN   
1         8/09/2007             15.0            >10   
2        11/12/2020              1.0              1   
3         3/10/2008             14.0            >10   
4        28/09/1984              NaN            NaN   
..              ...              ...            ...   
733      31/01/2022              0.0             <1   
734      11/07/2017              5.0              5   
735             NaN              NaN            NaN   
736             NaN              NaN            NaN   
737       9/09/2020              1.0              1   

    Principal actividad económica del negocio o empresa  \
0                      CULTURAL - GRUPO MUSICA LLANERA    
1    Venta de productos en cuero - Especialidad Sil...    
2                                     HOGAR GETRIATICO    
3                       ARTESANÍAS Y MUEBLES EN MIMBRE    
4                         Comercializadora de aluminio    
..                                                 ...    
733                                      comida rapida    
734                      SERVICIOS DE RECURSOS HUMANOS    
735     Comercialización de pijamas para mujer y niños    
736              servicios de mantenimiento industrial    
737                            Servicio de restaurante    

    Principal actividad económica del negocio o empresa V2 SECCION_COR  \
0                                        Grupo musical               R   
1                          Venta de productos en cuero               S   
2                                     Hogar geriátrico               Q   
3                                            Mimbrería               C   
4                                    Venta de aluminio               G   
..                                                 ...             ...   
733                                Venta comida rápida               I   
734                                Operador logístico                N   
735                                      Venta de ropa               G   
736              Servicios de mantenimiento industrial               M   
737                                        Restaurante               I   

                DES_SECCION_COR  \
0        Actividades artísticas   
1              Ot

1.  Análisis general del dataset (size, shape, describe)

In [ ]:
### Revisión de la dimensión del dataset
print("Las dimensiones de la Base de datos es de:", df.shape)
print("El tamaño de la base es:", df.size)

Las dimensiones de la Base de datos es de: (738, 67)
El tamaño de la base es: 49446


El tamaño del dataset muestra que la encuesta se realizo a 738 empresarios y el cuestinario contó con 68 preguntas y por tanto el dataset cuenta con 50.814 elementos

In [ ]:
###Descripción de las variables String
pd.set_option('display.max_columns', None)
df.describe(include = 'object').transpose()

count unique  \
Municipio                                            738     13   
Sexo del propietario del negocio o empresa:          738      2   
Registrado CCC                                       738      2   
FECHA_MATRICULA                                      526    468   
AÑOS ANTIGU V2                                       524      9   
...                                                  ...    ...   
Congestión vehicular:¿Cuáles son los dos princi...   104      1   
Ruido:¿Cuáles son los dos principales problemas...    70      1   
Ninguno:¿Cuáles son los dos principales problem...   205      1   
Otro, ¿Cuál?:¿Cuáles son los dos principales pr...   107      1   
Otro, ¿Cuál?:¿Cuáles son los dos principales pr...   106     74   

                                                                     top freq  
Municipio                                                     El Poblado   92  
Sexo del propietario del negocio o empresa:                        Mujer  404  
Registrado CCC                                                        Sí  523  
FECHA_MATRICULA                                               25/08/2021    4  
AÑOS ANTIGU V2                                                        <1  114  
...                                                                  ...  ...  
Congestión vehicular:¿Cuáles son los dos princi...  Congestión vehicular  104  
Ruido:¿Cuáles son los dos principales problemas...                 Ruido   70  
Ninguno:¿Cuáles son los dos principales problem...               Ninguno  205  
Otro, ¿Cuál?:¿Cuáles son los dos principales pr...                  Otro  107  
Otro, ¿Cuál?:¿Cuáles son los dos principales pr...    Estado de las vías    9  

[64 rows x 4 columns]

64 de las 68 columnas de la base son variables String

In [ ]:
###Descripción de las variables Númericas de la base
df.describe()

¿Qué edad tiene el propietario del negocio o empresa?  AÑOS_ANTIGUEDAD  \
count                                         738.000000           559.000000   
mean                                           44.941734            13.406082   
std                                            12.803096            29.056515   
min                                             0.000000             0.000000   
25%                                            35.000000             1.000000   
50%                                            44.000000             4.000000   
75%                                            54.000000            11.000000   
max                                            86.000000           122.000000   

       ¿Cuántas personas trabajan en su negocio o empresa (incluido usted como propietario)?  
count                                         738.000000                                      
mean                                            3.078591                                      
std                                             4.256600                                      
min                                             1.000000                                      
25%                                             1.000000                                      
50%                                             2.000000                                      
75%                                             3.000000                                      
max                                            70.000000

Las 3 columnas restantes son numéricas y son:

**Edad del propietario:** en promedio es 44 años, el 50% de los empresarios encuestados tienen entre 35 y 54 años (q1 y q3)

**Años de antigüedad de la Empresa:** la Media es de 13 años, y la mediana es de 4 años, pero se ve que el registro máximo de 122 años este dato atípico está afectado la distribución de esta variable

**Número de Empleados del Negocio:** en promedio las empresas encuestadas cuentan con 3 empleados, y la mediana es de 2 empleados, nuevamente el valor máximo de esta variable en un dato atípico que afecta el cálculo de las medidas de tendencia central, una empresa con 70 empleados
 

2. Análisis de datos nulos

In [ ]:
df.isnull().sum().sort_values(ascending=False)

Página web:¿Con qué frecuencia utiliza las siguientes herramientas digitales para la venta de sus productos?            734
Otra, ¿Cuál?:¿Qué herramienta utiliza para llevar su registro de ventas y gastos?                                       729
Otro, ¿Cuál?:¿Estaría interesado en alguno de los siguientes programas y servicios de la CCC?:                          728
Otro, ¿Cuál?:¿Estaría interesado en alguno de los siguientes programas y servicios de la CCC?v2:                        728
Gota-Gota:Señale las dos principales fuentes de financiamiento de su negocio o empresa?                                 719
                                                                                                                       ... 
Tik Tok:¿Con qué frecuencia utiliza las siguientes herramientas digitales para la venta de sus productos?                 0
Correo electrónico:¿Con qué frecuencia utiliza las siguientes herramientas digitales para la venta de sus productos?      0
¿Conoce 